packages/transformers/webextension/src/WebExtensionTransformer.js

```diff
@@ -216,6 +216,26 @@ async function collectDependencies(
     }
     program.web_accessible_resources = war;
   }
+  if (program.declarative_net_request) {
+    const rrs: {|path: string, id: string, enabled: boolean|}[] =
+      program.declarative_net_request?.rule_resources ?? [];
+    rrs.forEach((resources, i) => {
+      const loc = {
+        filePath,
+        ...getJSONSourceLocation(
+          ptrs[`/declarative_net_request/rule_resources/${i}/path`],
+          'value',
+        ),
+      };
+
+      asset.addURLDependency(resources.path, {
+        needsStableName: true,
```

Comment:
Sorry for not adding this earlier, but why does this need a stable name? We typically only reserve that for files that actually cannot move locations according to the web extension documentation, i.e. `_locales/en/messages.json`.

Reply:
Unless the transformer can change the path in the manifest to the destination of the transformed json files, it needs to exist at the same path. The rules cannot be inlined.

Reply:
The replacement is done in the packager automatically. This is the reason you see hashes in the filenames for other built files in the manifest like background scripts - their paths are rewritten in the transformer and fixed in the packager. This technique is useful for caching purposes and avoiding name collisions.

Reply:
Ah right, it's because you aren't using the return value from `addURLDependency`. Just set `rule_resources[i].path = asset.addURLDependency(...)`.

Reply:
Done. Sorry for the confusion.